src/main/java/com/minecraftabnormals/abnormals_core/core/config/ACConfig.java

```diff
@@ -1,22 +1,30 @@
 package com.minecraftabnormals.abnormals_core.core.config;
 
+import com.minecraftabnormals.abnormals_core.core.annotations.ConfigKey;
 import org.apache.commons.lang3.tuple.Pair;
 
 import net.minecraftforge.common.ForgeConfigSpec;
 import net.minecraftforge.common.ForgeConfigSpec.ConfigValue;
 import net.minecraftforge.fml.ModList;
 import net.minecraftforge.fml.config.ModConfig;
 
+import java.util.function.Predicate;
+
 /**
  * @author SmellyModder(Luke Tonon)
  */
 public final class ACConfig {
 
 	public static class Common {
+		@ConfigKey("quark_sign_editing_enabled")
 		public final ConfigValue<Boolean> enableQuarkSignEditing;
+		@ConfigKey("sign_editing_requires_empty_hand")
 		public final ConfigValue<Boolean> signEditingRequiresEmptyHand;
+		@ConfigKey("quark_poison_potato_compat_enabled")
 		public final ConfigValue<Boolean> poisonPotatoCompatEnabled;
+		@ConfigKey("potato_poison_effect")
 		public final ConfigValue<Boolean> poisonEffect;
+		@ConfigKey("potato_poison_chance")
 		public final ConfigValue<Double> poisonChance;
```

Comment:
I noticed that only the fields in `SlabfishSettings` were being looked up. I think it'd be good if these common values were also looked up for possible future use.
I'm also not really a fan of the `enabled` suffix to some of the `@ConfigKey`s

Reply:
What do you mean by 'looked up'? The enabled suffix isn't essential but I thought it would make the JSON clearer.

Reply:
It's true that `ValuesHolder` is used to look up config values from within the code, but I think referring directly to the values in JSON should be fine as most of the methods are just straight field returns. There is one that checks if quark's loaded first, but that can be done with the `forge:mod_loaded` predicate